src/sql/src/plan/query.rs

```diff
@@ -5586,15 +5590,50 @@ impl Visit<'_, Aug> for WindowFuncCollector {
     }
 }
 
-/// Specifies how long a query will live. This impacts whether the query is
+/// Specifies how long a query will live.
+///
+/// Currently, it affects whether SHOW commands are allowed.
+///
+/// (This used to also impact whether the query is
 /// allowed to reason about the time at which it is running, e.g., by calling
-/// the `now()` function.
+/// the `now()` function. Nowadays, this is decided by a different mechanism.)
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 pub enum QueryLifetime {
-    /// The query's result will be computed at one point in time.
+    /// The query's (or the expression's) result will be computed at one point in time.
     OneShot,
-    /// The query's result will be maintained indefinitely.
-    Static,
+    /// The query (or expression) is used in a dataflow that maintains an index.
+    Index,
+    /// The query (or expression) is used in a dataflow that maintains a materialized view.
+    MaterializedView,
+    /// The query (or expression) is used in a dataflow that maintains a SUBSCRIBE.
+    Subscribe,
+    /// The query (or expression) is part of a (non-materialized) view.
+    View,
+    /// The expression is part of a source definition.
+    Source,
+}
+
+impl QueryLifetime {
+    pub fn is_one_shot(&self) -> bool {
+        let res = matches!(self, QueryLifetime::OneShot);
```

Comment:
All these methods should use match blocks without `_` so any changes in the future must be thought about.

Reply:
Done